### ISSUE TITLE
Update to 0.3.0 breaks example 

### ISSUE THREAD

**Issue by jhnnsrs** [JaneliaSciComp/xarray-multiscale#10]
Create utility functions that I rely on a daily basis!

The 0.3.0 update breaks the listed example though:
Seems like the api for the reduction function has changed.

```python
from xarray_multiscale import multiscale
import numpy as np

data = np.arange(4)
x = multiscale(data, np.mean, (2,))
print(x)

```

```
Traceback (most recent call last):
  File "tt.py", line 5, in <module>
    x = multiscale(data, np.mean, (2,))
  File "/usr/local/lib/python3.8/site-packages/xarray_multiscale/multiscale.py", line 131, in multiscale
    downscaled = downscale(source, reduction, scale, pad_mode=pad_mode)
  File "/usr/local/lib/python3.8/site-packages/xarray_multiscale/multiscale.py", line 294, in downscale
    downscaled_data = downscale_dask(
  File "/usr/local/lib/python3.8/site-packages/xarray_multiscale/multiscale.py", line 278, in downscale_dask
    meta = reduction(
  File "<__array_function__ internals>", line 5, in mean
  File "/usr/local/lib/python3.8/site-packages/numpy/core/fromnumeric.py", line 3440, in mean
    return _methods._mean(a, axis=axis, dtype=dtype,
  File "/usr/local/lib/python3.8/site-packages/numpy/core/_methods.py", line 167, in _mean
    rcount = _count_reduce_items(arr, axis, keepdims=keepdims, where=where)
  File "/usr/local/lib/python3.8/site-packages/numpy/core/_methods.py", line 76, in _count_reduce_items
    items *= arr.shape[mu.normalize_axis_index(ax, arr.ndim)]
numpy.AxisError: axis 2 is out of bounds for array of dimension 1

```

pip dump:
`
dask==2021.11.2
numcodecs==0.9.1
numpy==1.21.4
s3fs==2021.11.1
xarray==0.17.0
xarray-multiscale==0.3.0
zarr==2.8.3
`

Example still working in 0.2.0 :)
Best

**Comment by d-v-b** [comment]
Yes, I'm sorry about that. (I thought I was the only user.... )

For reductions I was using the dask corasen method directly. This method takes a function like `np.mean` and wraps it in functionality that accomplishes windowed averaging. However, working with functions that don't have the same API as np.mean is tough with this approach. So now, `multiscale` takes reduction functions that directly do windowed downscaling, where the first argument is an N-dimensional array, and the second argument is a tuple of scale factors.

I included two reduction functions that work in the new style here: https://github.com/JaneliaSciComp/xarray-multiscale/blob/main/src/xarray_multiscale/reducers.py#L6

I can update the documentation to reflect the new way of doing things. And thanks for opening the issue!

**Comment by jhnnsrs** [comment]
A typed well documented scientific library? You are certainly not the only user ;)

I was able to find the reduction method and it works like a charm once you modify

```
def downscale(
    array: DataArray,
    reduction: Callable[[NDArray[Any], Tuple[int, ...]], NDArray[Any]],
    scale_factors: Sequence[int],
    pad_mode: str,
    **kwargs: Any,
) -> Any:

    to_downscale = normalize_array(array, scale_factors, pad_mode=pad_mode)
    downscaled_data = downscale_dask(
        to_downscale.data, reduction, scale_factors, **kwargs
    )
    downscaled_coords = downscale_coords(to_downscale, scale_factors)
    return DataArray(
        downscaled_data, downscaled_coords, attrs=array.attrs, **dims=array.dims**
    )
```

The original code just infered the dims from the coords, which according to a beautiful error was deprecated
and is no longer supported for this xarray (0.17.0) version. Do you want me to right a pull request?

Thanks for maintainig :)

**Comment by d-v-b** [comment]
yes, a PR would be greatly appreciated!

**Comment by jhnnsrs** [comment]
#11 here it is ;)

**Comment by d-v-b** [comment]
fixed in #11
